nova_vm/src/ecmascript/builtins/control_abstraction_objects/promise_objects/promise_abstract_operations/promise_group_record.rs

```diff
@@ -103,7 +112,64 @@ impl<'a> PromiseGroup<'a> {
         }
     }
 
-    pub(crate) fn reject(self, agent: &mut Agent, value: Value<'a>, gc: NoGcScope<'a, '_>) {
+    pub(crate) fn reject(
+        self,
+        agent: &mut Agent,
+        index: u32,
+        error: Value<'a>,
+        gc: GcScope<'a, '_>,
+    ) {
+        let promise_group = self.bind(gc.nogc());
+        let error = error.bind(gc.nogc());
+
+        let promise_group_record = promise_group.get_mut(agent);
+        let (result_array, promise_to_resolve) = promise_group_record.take_result_and_promise();
+
+        let elements = result_array.as_mut_slice(agent);
+        elements[index as usize] = Some(error.unbind());
+
+        if let Some(promise_to_resolve) = promise_to_resolve {
+            let aggregate_error = OrdinaryObject::create_object(
+                agent,
+                Some(
+                    agent
+                        .current_realm_record()
+                        .intrinsics()
+                        .aggregate_error_prototype()
+                        .into(),
+                ),
+                &[ObjectEntry::new_data_entry(
```

Comment:
The spec also says that I should use `DefinePropertyOrThrow` when setting this. I hope this is the equivalent.